app/Main.hs

```diff
@@ -1,6 +1,37 @@
 module Main where
 
-import Lib
+import Control.Monad (join)
+
+import Options.Applicative
+
+import Cache (cacheBoards)
+import Search (searchBoards)
+
+data Command
+    = Search (Maybe String)
+    | Cache
 
 main :: IO ()
-main = someFunc
+main = run =<< execParser
+    (parseCommand `withInfo` "Interact with the Trello API")
+
+run :: Command -> IO ()
+run cmd =
+    case cmd of
```

Comment:
Similar to my reasonings above, this seems easier to parse and like less work as the number of commands grows... but I'm not convinced. I'm going to keep as is for now.